### PR TITLE
explicit error in case of misuse of dg.plug, fix #36

### LINKED ISSUE #36: Add check that `dg.plug` connects two signals
When using `dynamic_graph.plug(a, b)`, a common bug is to pass an operator for `a` or `b` instead of a signal. It would be great if the underlying `plug(...)` function asserts the passed in objects derive from `SignalBase`.

Here is a python hack that helped me to track down a plugging bug:

```python
import dynamic_graph as dg
plug_ = dg.plug

# Monkey patch it till you make it.
def dg_plug_dbg(sig0, sig1):
    def assert_signal(sig):
        if not isinstance(sig, dg.signal_base.SignalBase):
            raise ValueError('dynamic_graph.plug(): Passed in value is not a signal. sig=' + str(sig), sig)
    assert_signal(sig0)
    assert_signal(sig1)
    plug_(sig0, sig1)

dg.plug = dg_plug_dbg
```

\cc @MaximilienNaveau

### PR DESCRIPTION
**Pull request by nim65s**
While here, fix tests for python 3.8+

### CODE REVIEW

**Comment by nim65s** [comment]
@jviereck : would this work for you ?

File: src/dynamic_graph/dynamic-graph-py.cc [hunk 1]
```diff
@@ -40,10 +46,22 @@ PyObject* plug(PyObject* /*self*/, PyObject* args) {
 
   pObjIn = PyCapsule_GetPointer(objIn, "dynamic_graph.Signal");
   SignalBase<int>* signalIn = (SignalBase<int>*)pObjIn;
-  if (signalIn == NULL) return NULL;
+  if (signalIn == NULL) {
+      struct module_state* st = GETSTATE(m);
+      std::ostringstream oss;
+      oss << "dgpy.plug in argument must be a dynamic_graph.Signal, not a " << PyCapsule_GetName(objIn);
```
Comment (jviereck) [inline src/dynamic_graph/dynamic-graph-py.cc]: How about `dynamic_graph.plug(a, b): Argument 'a' must be of type 'dynamic_graph.Signal', but got 'xyz'`?

The string you propose is not wrong, but not everyone might understand what an "in argument" means here. Also, people usually do `import dynamic_graph` and then `dynamic_graph.plug`, so in that case, it's maybe not clear what `dgpy` means.

I try to design strings to be as much useful and understandable from the users perspective (and the knowledge/understanding the user is expected to have). 

Does that make sense?

**Comment by jviereck** [comment]
> @jviereck : would this work for you ?

It would work, proposed an alternative with the comment inline.

**Comment by jviereck** [comment]
Thanks for working on this @nim65s !

**Comment by jviereck** [comment]
Not tested but looks good for me now! Thanks.

**Comment by nim65s** [comment]
Thanks for your feedback !

File: src/dynamic_graph/dynamic-graph-py.cc [hunk 2]
```diff
@@ -40,10 +46,24 @@ PyObject* plug(PyObject* /*self*/, PyObject* args) {
...
+  }
   pObjOut = PyCapsule_GetPointer(objOut, "dynamic_graph.Signal");
   SignalBase<int>* signalOut = (SignalBase<int>*)pObjOut;
-  if (signalOut == NULL) return NULL;
+  if (signalOut == NULL) {
+      struct module_state* st = GETSTATE(m);
+      std::ostringstream oss;
+      oss << "dynamic_graph.plug(a, b): Argument 'a' must be of type 'dynamic_graph.Signal', but got "
+          << PyCapsule_GetName(objOut);
+      PyErr_SetString(st->dgpyError, oss.str().c_str());
```
Comment (jmirabel) [inline src/dynamic_graph/dynamic-graph-py.cc]: Why not `PyErr_SetString(PyExc_TypeError, oss.str().c_str());` ?

Comment (nim65s) [reply src/dynamic_graph/dynamic-graph-py.cc]: That would be better, indeed.

**Comment by nim65s** [comment]
So I exposed `dgpyError` for nothing, but this might be handful, so I guess we can keep it as a bonus.

**Pull request merged**